docs/application/dotnet/guides/nui/styling-controls-with-xaml.md

```diff
@@ -29,51 +62,115 @@ In addition, styles can also be defined for the components with the following st
 | `SelectedFocused`  | Specifies the state where the widget is checked and all events are received.                                                                                            |
 | `DisabledSelected` | Specifies the combination of disabled and selected states.                                                                                                       |
 
-The NUI also defines the `ViewStyle` derived classes for components such as `ImageView`, `TextLabel` and `TextField`:
+Some attributes are `Selector` type which express the change of values in differenct control state:
 
 ```csharp
-ImageViewStyle swtichImageStyle = new ImageViewStyle
+ControlStyle pressableStyle = new ControlStyle()
 {
-    ResourceURL = new StringSelector
-    {
-        Normal = "controller_switch_bg_off.png",
-        Selected = "controller_switch_bg_on.png",
-        Disabled = "controller_switch_bg_off_dim.png",
-        DisabledSelected = "controller_switch_bg_on_dim.png"
-    }
+  Size = new Size(100, 100),
+  BackgroundColor = new Selector<Color>
+  {
+      Normal = Color.White,
+      Pressed = Color.Yellow,
+      Disabled = Color.Grey
+  }
 };
 
-TextLabelStyle redTextStyle = new TextLabelStyle()
+Control control = new Control();
+control.ApplyStyle(pressableStyle);
+```
+
+The selectors work when the control state is enabled in a `View`. By default, the control state is enabled for all controls in the `Tizen.NUI.Components` namespace and not for the controls that belong to `Tizen.NUI.BaseComponents` namespace, such as `View` and `TextLabel`. If you want to enable a control state for them, set the `EnableControlState` property to `true`:
+
+```csharp
+View view = new View()
 {
-  TextColor = Color.Red,
-  PointSize = 20
+  EnableControlState = true,
 };
+```
 
-TextFieldStyle textStyle = new TextFieldStyle()
+## Theme
+
+`Theme` is a set of `ViewStyle` that helps to decorate an application.
+Here is an example of creating and applying a theme:
+
+```csharp
+// Define a theme
+var theme = new Theme();
+
+theme.AddStyle("BlueStyle", new ViewStyle() {
+  BackgroundColor = Color.Blue,
+});
+
+theme.AddStyle("YellowTextStyle", new TextLabelStyle() {
+  TextColor = Color.Yellow,
+  PointSize = 10
+});
+
+// Apply the theme to the current application.
+ThemeManager.ApplyTheme(theme);
+
+// Create views with style name
+var view = new View()
+{
+  StyleName = "BlueStyle",
+};
+
+var text = new TextLabel()
 {
-  EnableCursorBlink = False
+  StyleName = "YelloTextStyle"
 };
 ```
 
+Each style in a theme is identified by a style name and it will be matched the `View.StyleName`.
+When you set a style name to a full class name such as "Tizen.NUI.BaseComponents.TextLabel", it will apply to all instances of that type in an applciation:
```

Comment:
```suggestion
Each style in a theme is identified by a style name and it will be matched the `View.StyleName`.
When you set a style name to a full class name such as `Tizen.NUI.BaseComponents.TextLabel`, it will apply to all instances of that type in an application:

Reply:
Done.